sphinxcontrib/writers/rst.py

```diff
@@ -711,10 +711,7 @@ def visit_reference(self, node):
         Finally, all other links are also converted to an inline link
         format.
         """
-        if 'name' not in node:
-            self.add_text('`%s`_' % node.astext())
-            raise nodes.SkipNode
-        elif 'refuri' not in node:
+        if 'refuri' not in node and 'name' in node:
```

Comment:
This breaks the case if node contains a `refuri`, but no `name`. E.g. in this line:

    Direct hyperlinks, like http://www.python.org.

After this commit, it will take the if-branch `if 'internal' not in node:`, but that line `'%s <%s>_' % (node['name'], node['refuri'])` will fail with a KeyError because `node['name']` is undefined.

I was just working on this testcase :). I committed it to a3d8b67 at macfreek/restbuilder.

I will merge these commits and uses cases later when time permits.

Note that a3d8b67 is a commit in a branch called misc-improvements, which will likely be cherry-picked and deleted later. Feel free to cherry-pick, but please don't fork that branch.